test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -54,101 +65,156 @@ public class PassFailJFrame {
         } else {
-            invokeAndWait(() -> {
-                createUI(title, instructions, maxRowLength, maxStringLength, timeoutMinutes);
-            });
+            invokeAndWait(() -> createUI(title, instructions, rows, columns,
+                    testTimeOutInMinutes));
         }
     }
 
     private static void createUI(String title, String instructions,
-                                 int maxRowLength, int maxStringLength,
-                                 int timeoutMinutes) {
+                                 int rows, int columns,
+                                 int timeoutInMinutes) {
         frame = new JFrame(title);
         frame.setLayout(new BorderLayout());
-        instructionsText = new JTextArea(instructions, maxRowLength, maxStringLength);
+        JTextArea instructionsText = new JTextArea(instructions, rows, columns);
         instructionsText.setEditable(false);
-        instructionsText.setFocusable(false);
-        frame.add(instructionsText, BorderLayout.NORTH);
+        instructionsText.setLineWrap(true);
+
+        int testTimeout = (int) TimeUnit.MINUTES.toMillis(timeoutInMinutes);
+
+        final JLabel testTimeoutLabel = new JLabel(String.format("Test " +
+                "timeout: %s", convertMillisToTimeStr(testTimeout)), JLabel.CENTER);
+        final long startTime = System.currentTimeMillis();
+        timer.setDelay(1000);
+        timer.addActionListener((e) -> {
+            int leftTime = testTimeout - (int) (System.currentTimeMillis() - startTime);
+            if ((leftTime < 0) || failed) {
+                timer.stop();
+                testFailedReason = "Failure Reason:\n Timeout " +
+                        "User did not perform testing.";
+                timeout = true;
+                latch.countDown();
+            }
+            testTimeoutLabel.setText(String.format("Test timeout: %s", convertMillisToTimeStr(leftTime)));
+        });
+        timer.start();
+        frame.add(testTimeoutLabel, BorderLayout.NORTH);
+        frame.add(new JScrollPane(instructionsText), BorderLayout.CENTER);
 
         JButton btnPass = new JButton("Pass");
-        btnPass.addActionListener((e) -> latch.countDown());
+        btnPass.addActionListener((e) -> {
+            latch.countDown();
+            timer.stop();
+        });
 
         JButton btnFail = new JButton("Fail");
-        btnFail.addActionListener((e) -> getFailureReason());
+        btnFail.addActionListener((e) -> {
+            getFailureReason();
+            timer.stop();
+        });
 
         JPanel buttonsPanel = new JPanel();
         buttonsPanel.add(btnPass);
         buttonsPanel.add(btnFail);
 
+        frame.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                super.windowClosing(e);
+                testFailedReason = "Failure Reason:\n User closed the " +
+                        "instruction Frame";
```

Comment:
```suggestion
                testFailedReason = "Failure Reason:\n"
                        + "User closed the instruction frame";
```
I think it's easier to read this way, the parts of the phrase aren't broken.